sidecar/tests/test_accel.py

```
@@ -164,34 +164,62 @@ def test_vram_gate_reads_vendor_agnostic_free(monkeypatch):
     assert notice and "CPU" in notice
 
 
+def test_vram_gate_keeps_metal_on_unified_memory(monkeypatch):
+    # Apple silicon: CPU and Metal share one pool, so the proactive gate
+    # must not demote a Metal plan that looks too big for "free" — that
+    # frees nothing and loses the accelerator (planner's unified-memory rule).
+    monkeypatch.setattr(accel, "device_free_bytes", lambda: 2 * _GIB)
+    monkeypatch.setattr(accel, "_model_weight_bytes", lambda a: 5 * _GIB)
+    attempted = []
+    class FakeBackend:
+        def load(self, a, device, ct, config=None): attempted.append(device); self.loaded = True
```

Ruff (0.16.3)

[error] Multiple statements on one line (semicolon)

(E702)